arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowAdapterTest.java

```diff
@@ -0,0 +1,503 @@
+        + "  ArrowProject(intField=[$0], stringField=[$1])\n"
+        + "    ArrowTableScan(table=[[ARROW, ARROWDATA]], fields=[[0, 1, 2, 3]])\n\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .limit(6)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  @Test void testArrowProjectFieldsWithIntegerFilter() {
+    String sql = "select \"intField\", \"stringField\"\n"
+        + "from arrowdata\n"
+        + "where \"intField\" < 4";
+    String result = "intField=0; stringField=0\n"
+        + "intField=1; stringField=1\n"
+        + "intField=2; stringField=2\n"
+        + "intField=3; stringField=3\n";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowProject(intField=[$0], stringField=[$1])\n"
+        + "    ArrowFilter(condition=[<($0, 4)])\n"
+        + "      ArrowTableScan(table=[[ARROW, ARROWDATA]], fields=[[0, 1, 2, 3]])\n\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .limit(4)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  @Test void testArrowProjectFieldsWithMultipleFilters() {
+    String sql = "select \"intField\", \"stringField\"\n"
+        + "from arrowdata\n"
+        + "where \"intField\"=12 and \"stringField\"='12'";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowProject(intField=[$0], stringField=[$1])\n"
+        + "    ArrowFilter(condition=[AND(=($0, 12), =($1, '12'))])\n"
+        + "      ArrowTableScan(table=[[ARROW, ARROWDATA]], fields=[[0, 1, 2, 3]])\n\n";
+    String result = "intField=12; stringField=12\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .limit(3)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  @Test void testArrowProjectFieldsWithFloatFilter() {
+    String sql = "select * from arrowdata\n"
+        + " where \"floatField\"=15.0";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowFilter(condition=[=(CAST($2):DOUBLE, 15.0)])\n"
+        + "    ArrowTableScan(table=[[ARROW, ARROWDATA]], fields=[[0, 1, 2, 3]])\n\n";
+    String result = "intField=15; stringField=15; floatField=15.0; longField=15\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  @Test void testArrowProjectFieldsWithFilterOnLaterBatch() {
+    String sql = "select \"intField\"\n"
+        + "from arrowdata\n"
+        + "where \"intField\"=25";
+    String plan = "PLAN=ArrowToEnumerableConverter\n"
+        + "  ArrowProject(intField=[$0])\n"
+        + "    ArrowFilter(condition=[=($0, 25)])\n"
+        + "      ArrowTableScan(table=[[ARROW, ARROWDATA]], fields=[[0, 1, 2, 3]])\n\n";
+    String result = "intField=25\n";
+    CalciteAssert.that()
+        .with(arrow)
+        .query(sql)
+        .returns(result)
+        .explainContains(plan);
+  }
+
+  // TODO: test a table whose field names contain spaces,
```

Comment:
From a cursory look, it seems that we should add at least all the tests associated with these TODOs, and possibly others.

Reply:
There seems to be consensus in the [Jira ticket](https://issues.apache.org/jira/browse/CALCITE-2040) to get the consolidated version of what we currently have in, and file tickets for the current bugs and limitations. 

So we can either implement those tests and disable them if not working (and of course filing an associated ticket for "family" of failing tests, like "add support for queries over multiple tables" etc.), or directly file a ticket and add what's in the TODO as part of the description, so we don't lose the context those comments provide, before deleting them.

Let's see what will be better when we reach a more mature state.

Reply:
I have done an improvement pass on the tests (see [a90ddb8](https://github.com/asolimando/calcite/commit/a90ddb8bac43ac1716f46716d47de5e9a431c902)), for the remaining tests I would do the same (try to implement them, and disabled the test if not currently working)

Reply:
@macroguo-ghy could you file an additional ticket for improving test coverage and adding the different tests marked here as TODO in the description, and remove them from the file?

Because we don't really use TODOs in Calcite, but the suggested tests are good, so it would be a pity if we lost track of them.